packages/rule-utils/src/variants.ts

```diff
@@ -12,13 +12,17 @@ export function variantMatcher(name: string, handler: (input: VariantHandlerCont
 
       const match = input.match(re)
       if (match) {
-        return {
-          matcher: input.slice(match[0].length),
+        const matcher = input.slice(match[0].length)
+        const handlers: VariantHandler[] = toArray(handler).map(handler => ({
+          matcher,
           handle: (input, next) => next({
             ...input,
             ...handler(input),
           }),
-        }
+        }))
+        return handlers.length === 1
+          ? handlers[0]
+          : handlers
```

Comment:
For backward compat reasons, this ensures that changes from this PR don't affect the function's output if the input doesn't change. Always returning arrays broke one of our tests so it looks like it could also happen on user code.